### ISSUE TITLE
Missing @id in schema.org/Article example

### ISSUE THREAD

**Issue by jvandriel** [schemaorg/schemaorg#906]
I've noticed the third JSON-LD example on http://schema.org/Article is missing an @id on:

 `"isPartOf": "#issue",`

The JSON-LD specs say about this: _"In JSON-LD, a node is identified using the @id keyword"_ (http://www.w3.org/TR/json-ld/#node-identifiers)

As illustrated by both Google's Structured Data Validator and the Structured Data Linter not extracting the right information from the current example.

The code should therefore be changed to:

```
"isPartOf":{
    "@id":"#issue"
}
```

Furthermore I was wondering if there's an easy way of checking whether this is happening in any other example on schema.org?

**Comment by gkellogg** [comment]
In the absence of `@value` on an object, the object is also considered a node, with a blank node `@id`, so:

```json
"isPartOf": {
  "name": "Something"
}
```

is effectively equivalent to the following:

```json
"isPartOf": {
  "@id": "_:something",
  "name": "Something"
}
```

However, good Linked Data principles suggest the use of IRIs to identify things, even embedded nodes, so using a fragment identifier as above is a best practice, and should be encouraged for schema.org examples, IMO.

**Comment by danbri** [comment]
Can we set aside the Linked Data obsession with requiring URIs for everything (which schema.org does not as a project historically share). 

Trying the example in json-ld playground, it gives:  _:b0 <http://schema.org/isPartOf> "#issue" . ... rather than joining up to the URI used elsewhere in the data. 

If we want "isPartOf": "#issue" ...to work this way we'll need to have it listed as taking @id in our @context file (see http://schema.org/docs/jsonldcontext.json ). I believe the current code for that list uses the hint that any property whose rangeIncludes URL is treated as by default an @id property for JSON-LD purposes. So we could do something like that, or add another internal mechanism for listing which properties default to @id behaviour.

**Comment by jvandriel** [comment]
Did @danbri's suggestion ever get implemented? 

Reason I ask is because neither the SDTT nor Structured Data Linter still seem to be able to parse (the now 4th example) successfully.

**Comment by danbri** [comment]
Just to make sure we're talking about the same example,

```
<script type="application/ld+json">
{
  "@context": "http://schema.org",
  "@graph": [
    {
        "@id": "#issue",
        "@type": "PublicationIssue",
        "issueNumber": "5",
        "datePublished": "2012",
        "isPartOf": {
            "@id": "#periodical",
            "@type": [
                "PublicationVolume",
                "Periodical"
            ],
            "name": "Cataloging & Classification Quarterly",
            "issn": [
                "1544-4554",
                "0163-9374"
            ],
            "volumeNumber": "50",
            "publisher": "Taylor & Francis Group"
        }
    },
    {
        "@type": "ScholarlyArticle",
        "isPartOf": "#issue",
        "description": "The library catalog as a catalog of works was an infectious idea, which together with research led to reconceptualization in the form of the FRBR conceptual model. Two categories of lacunae emerge--the expression entity, and gaps in the model such as aggregates and dynamic documents. Evidence needed to extend the FRBR model is available in contemporary research on instantiation. The challenge for the bibliographic community is to begin to think of FRBR as a form of knowledge organization system, adding a final dimension to classification. The articles in the present special issue offer a compendium of the promise of the FRBR model.",
        "sameAs": "https://doi.org/10.1080/01639374.2012.682254",
        "about": [
            "Works",
            "Catalog"
        ],
        "pageEnd": "368",
        "pageStart": "360",
        "name": "Be Careful What You Wish For: FRBR, Some Lacunae, A Review",
        "author": "Smiraglia, Richard P."
    }
  ]
}
</script>
```

**Comment by danbri** [comment]
In schema.org JSON-LD context file: 

>         "isPartOf": { "@id": "schema:isPartOf", "@type": "@id"},

**Comment by jvandriel** [comment]
Yep, that's the correct example indeed. I just checked it with the Rich Result Test, Structured Data Testing Tool, Structured Data Linter and JSON-LD playground and none of them join the pieces together.
